adb_graphics/datahandler/grib.py

```diff
@@ -210,19 +330,24 @@ def get_transform(self, transforms, val):
 
     @property
     def grid_suffix(self):
-        ''' Return the suffix of the first variable in the file. This should
-        correspond to the grid tag. '''
 
-        var = list(self.ds.keys())[0]
-        return var.split('_')[-1]
+        ''' Return the suffix of the first variable with 4 sections (split on _)
+        in the file. This should correspond to the grid tag. '''
+
+        for var in self.ds.keys():
+            vsplit = var.split('_')
+            if len(vsplit) == 4:
+                return vsplit[-1]
+        return 'GRID NOT FOUND'
 
 
     def latlons(self):
 
         ''' Returns the set of latitudes and longitudes '''
 
-        return [self.ds.coords[c].values for c in \
-                list(self.ds.coords)[-2:]]
+        coords = sorted([c for c in list(self.ds.coords) if
+                         any(ele in c for ele in ['lat', 'lon'])])
+        return [self.ds.coords[c].values for c in coords]
```

Comment:
With all the combining of datasets happening now, there was no set order to expect for finding latitude and longitude. Hopefully all types of files will have lat or lon as part of the coordinate name.

Reply:
NCL had the capability of generating the lat, lon arrays when reading gridded data.  I would imagine that PyNio has this same capability, and depending on the grid, the arrays were named lat_0, lon_0, gridlat_0, gridlon_0, so I think you're good, here.

Reply:
We are no longer using PyNio as the data handler, but all of these are XArrays with naming conventions provided by PyNio. This is the function that replicates that capability so that we can extract latitude and longitude arrays.